app/web/src/components/IdeToolbarNew/IdeToolbarNew.js

```diff
@@ -5,18 +5,34 @@ import { useIdeState, codeStorageKey } from 'src/helpers/hooks/useIdeState'
 import { copyTextToClipboard } from 'src/helpers/clipboard'
 import { requestRender } from 'src/helpers/hooks/useIdeState'
 import { encode, decode } from 'src/helpers/compress'
+import { flow } from 'lodash/fp'
+
+export const githubSafe = (url) =>
+  url.includes('github.com')
+    ? url
+        .replace('github.com', 'raw.githubusercontent.com')
+        .replace('/blob/', '/')
+    : url
+
+const prepareEncodedUrl = flow(decodeURIComponent, githubSafe)
```

Comment:
Trying to ease my way into some functional concepts. This is just a helper that makes a function the equivalent of 
```
url => githubSafe(decodeURIComponent(url))
```
Not really needed in this case with only two functions, but the idea is one of the dis-incentives of making small easy to understand functions is that chaining them is pain to write and read `a(b(c(input)))` so you just make one `abc(input)`.